vignettes/forecast_data.Rmd

```diff
@@ -0,0 +1,265 @@
+)
+```
+
+For purposes of evaluating predictions, it can be helpful to join the observed target values, contained in `forecast_target_observations`, into the data frame of forecast outputs. This enables direct comparison of predictions and observations. We illustrate this here, omitting some columns from the display for convenience:
+
+```{r}
+forecast_outputs |>
+  left_join(forecast_target_observations) |>
+  select(-model_id, -reference_date, -horizon)
+```
+
+## The `wk flu hosp rate` target
+
+The `wk flu hosp rate` target represents the rate of weekly confirmed influenza hospital admissions per 100,000 population. Note that this target was not included in the FluSight hub; we have introduced it here for illustrative purposes. We have used population values of 6,978,662 for Massachusetts and 29,914,599 for Texas. These population values are sourced from the [auxiliary data file](https://github.com/cdcepi/FluSight-forecast-hub/blob/main/auxiliary-data/locations.csv) provided by the FluSight hub, which are also reproduced in the [example-complex-forecast-hub](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-forecast-hub) repository.
+
+For this target, we created cumulative distribution function (CDF) predictions with evenly spaced CDF evaluation points ranging from 0.25 to 25 in increments of 0.25 hospitalizations per 100,000 population:
+
+```{r}
+forecast_outputs |>
+  filter(target == "wk flu hosp rate") |>
+  select(-model_id, -reference_date, -horizon) |>
+  head()
+```
+
+For the CDF `output_type`, the `output_type_id` contains the value at which the predictive CDF was evaluated, and the `value` contains the predicted probability that the target is less than or equal to that evaluation point. In the above example, the `value` in the row with `output_type_id` equal to 1.5 contains the model's estimated probability that the rate of hospital admissions in Texas the week of December 17, 2022 would be less than or equal to 1.5 admissions per 100,000 population. Again, these CDF values were estimated from the original quantile forecasts using the methods in the `distfromq` package.
+
+The following plot illustrates the predictive CDFs produced by the `MOBS-GLEAM_FLUH` model for Massachusetts on the reference date 2022-12-17, with each `target_end_date` shown in a separate facet.  Also shown in orange is a CDF representing the observation for this target, which was between 9.75 and 10 hospitalizations per 100,000 population.  This CDF corresponds to a point mass at the observed value, with a value of 0 below the observation and a value of 1 above the observation.
+
+```{r}
+# Subset the forecasts and observations to those that we will plot
+forecasts_to_plot <- forecast_outputs |>
+  filter(
+    model_id == "MOBS-GLEAM_FLUH",
+    target == "wk flu hosp rate",
+    location == "25",
+    reference_date == "2022-12-17"
+  ) |>
+  mutate(output_type_id = as.numeric(output_type_id))
+head(forecasts_to_plot)
```

Comment:
It would be useful to be able to see the `output_type_id` and `value` columns here